pkgs/misc/vscode-extensions/rust-analyzer/default.nix

```
@@ -37,9 +39,15 @@ let
     inherit releaseTag;
 
     nativeBuildInputs = [
-      jq moreutils esbuild
+      jq
+      moreutils
+      esbuild
       # Required by `keytar`, which is a dependency of `vsce`.
-      pkg-config libsecret
+      pkg-config
+      libsecret
+    ] ++ lib.optional stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.AppKit
```

```suggestion      
      AppKit
      Security
```
and add those to inputs and inherit the frameworks in the top-level call.

---

Not 100% sure I did what you meant, please take a look :)